diff --git a/dist/server/middlewareHandler.js b/dist/server/middlewareHandler.js
index 006458cb010d5ed9f31f03a17acc2db3aa093462..73cebbe6412c56b5d03a264514eeb7dbfac665eb 100644
--- a/dist/server/middlewareHandler.js
+++ b/dist/server/middlewareHandler.js
@@ -9,11 +9,16 @@ function createClerkHandler(eventHandler, clerkOptions = {}) {
       try {
         const loadedOptions = loadOptions(request, clerkOptions);
         const requestState = await authenticateRequest(request, loadedOptions);
-        const clerkInitialState = getResponseClerkState(requestState, loadedOptions);
+        const clerkInitialState = getResponseClerkState(
+          requestState,
+          loadedOptions
+        );
         router.update({
-          context: clerkInitialState
+          context: {
+            ...router.options.context,
+            ...clerkInitialState,
+          },
         });
-        await router.load();
       } catch (error) {
         if (error instanceof Response) {
           return error;
